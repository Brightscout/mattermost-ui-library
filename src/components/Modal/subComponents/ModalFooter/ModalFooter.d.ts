import React from 'react';

export interface ModalFooterProps {

    /**
     * Primary action component
     */
    primary: React.ReactNode;

    /**
     * Secondary action component
     */
    secondary?: React.ReactNode;
}