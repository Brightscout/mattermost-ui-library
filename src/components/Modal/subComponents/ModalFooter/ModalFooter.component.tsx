import React from 'react';

import {ModalFooterProps} from './ModalFooter';
import {ModalFooterWrapper} from './ModalFooter.styles';

/**
 * ModalFooter component
 *
 * @example Correct usage
 * <ModalFooter
 *      primary={<Button variant='primary'>{'Submit'}</Button>}
 *      secondary={<Button variant='tertiary'>{'Close'}</Button>}
 * />
 */
export const ModalFooter = (props: ModalFooterProps) => {
    const {primary, secondary} = props;
    return (
        <ModalFooterWrapper className='mm=modalFooter'>
            {secondary}
            {primary}
        </ModalFooterWrapper>);
};