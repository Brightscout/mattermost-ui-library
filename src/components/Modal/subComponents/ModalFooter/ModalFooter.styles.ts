import styled from 'styled-components';

import Colors from '@Styles/colorsForJs.module.scss';

export const ModalFooterWrapper = styled.div({
    display: 'flex',
    justifyContent: 'flex-end',
    gap: '8px',
    padding: '24px 32px',
    borderTop: `1px solid ${Colors.centerChannel_16}`,
});