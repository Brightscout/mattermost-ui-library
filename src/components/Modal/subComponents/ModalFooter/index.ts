export {ModalFooter} from './ModalFooter.component';
export type {ModalFooterProps} from './ModalFooter';
