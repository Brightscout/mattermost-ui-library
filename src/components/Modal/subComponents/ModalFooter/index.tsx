import React from 'react';
import {Modal as RBModal} from 'react-bootstrap';

type ModalFooterProps = {
    onConfirm?: (() => void) | null;
    confirmBtnText?: string;
    cancelBtnText?: string;
    onHide?: () => void;
    className?: string;
    confirmDisabled?: boolean;
    cancelDisabled?: boolean;
    confirmBtnClassName?: string;
}

const ModalFooter = ({
    onConfirm,
    onHide,
    cancelBtnText,
    confirmBtnText,
    className = '',
    confirmDisabled,
    cancelDisabled,
    confirmBtnClassName = '',
}: ModalFooterProps) : JSX.Element => (
    <RBModal.Footer className={`modal__footer d-flex flex-row-reverse align-items-center ${className}`}>
        {onConfirm && (
            <button
                className={`btn modal__confirm-btn margin-left-10 ${confirmBtnClassName || 'btn-primary'}`}
                onClick={onConfirm}
                disabled={confirmDisabled}
            >
                {confirmBtnText || 'Confirm'}
            </button>
        )}
        {onHide && (
            <button
                className='btn btn-link modal__cancel-btn margin-0 padding-0'
                onClick={onHide}
                disabled={cancelDisabled}
            >
                {cancelBtnText || 'Cancel'}
            </button>
        )}
    </RBModal.Footer>
);

export default ModalFooter;
