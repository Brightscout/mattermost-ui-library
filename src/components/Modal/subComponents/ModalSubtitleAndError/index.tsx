import React from 'react';

type ModalSubtitleAndErrorProps = {
    subTitle?: string | JSX.Element;
    error?: string | JSX.Element;
}

const ModalSubtitleAndError = ({
    subTitle,
    error,
}: ModalSubtitleAndErrorProps) : JSX.Element => (
    <>
        {subTitle && <p className='modal__subtitle margin-0 margin-bottom-15'>{subTitle}</p>}
        {error && <p className='modal__error margin-left-15 margin-bottom-10 error-text text-capitalize'>{error}</p>}
    </>
);

export default ModalSubtitleAndError;
