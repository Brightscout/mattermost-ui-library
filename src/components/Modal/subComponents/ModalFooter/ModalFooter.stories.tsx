import React from 'react';
import {Story, Meta} from '@storybook/react';

import {ModalFooterProps} from './ModalFooter';
import {ModalFooter} from './ModalFooter.component';

export default {
    title: 'Component/Modal/ModalFooter',
    component: ModalFooter,
} as Meta<ModalFooterProps>;

const ModalFooterTemplate: Story<ModalFooterProps> = (args) => <ModalFooter {...args}/>;

// Default
export const Default = ModalFooterTemplate.bind({});
