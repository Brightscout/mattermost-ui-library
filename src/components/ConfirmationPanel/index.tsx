import React, {forwardRef} from 'react';

// Components
import ModalBody from '@Components/Modal/subComponents/ModalBody';
import ModalFooter from '@Components/Modal/subComponents/ModalFooter';
import ModalSubtitleAndError from '@Components/Modal/subComponents/ModalSubtitleAndError';

type ConfirmationPanelProps = {
    confirmationMsg: string;
    onHide: () => void;
    onConfirm?: (() => void) | null;
    loading?: boolean;
    className?: string;
    error?: string | JSX.Element;
}

const ConfirmationPanel = forwardRef<HTMLDivElement, ConfirmationPanelProps>(({
    confirmationMsg,
    onHide,
    onConfirm,
    loading,
    className = '',
    error,
}: ConfirmationPanelProps, confirmationPanelRef) => (
    <div
        ref={confirmationPanelRef}
        className={`confirmation-panel primary-panel ${className}`}
    >
        <ModalBody>
            <p className='margin-top-10 margin-h-0 margin-bottom-15'>{confirmationMsg}</p>
        </ModalBody>
        <ModalSubtitleAndError error={error}/>
        <ModalFooter
            onConfirm={onConfirm}
            onHide={onHide}
            confirmBtnText='Delete'
            cancelBtnText='Cancel'
            confirmDisabled={loading}
            cancelDisabled={loading}
            className='padding-v-12 padding-h-15'
        />
    </div>
));

export default ConfirmationPanel;
