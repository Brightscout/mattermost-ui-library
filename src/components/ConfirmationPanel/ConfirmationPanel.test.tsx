import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

import ModalBody from '@Components/Modal/subComponents/ModalBody';
import ModalFooter from '@Components/Modal/subComponents/ModalFooter';
import ModalSubtitleAndError from '@Components/Modal/subComponents/ModalSubtitleAndError';

// Mock data
import {confirmHandler, hideHandler, confirmationPanelProps} from '@Constants/mockData/confirmationPanel';

import ConfirmationPanel from '.';

describe('ConfirmationPanel', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<ConfirmationPanel {...confirmationPanelProps}/>);
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed "className" prop', () => {
        expect(component.hasClass(confirmationPanelProps.className)).toBeTruthy();
    });

    it('Should render the message correctly', () => {
        expect(component.contains(
            <ModalBody>
                <p className='margin-top-10 margin-h-0 margin-bottom-15'>{confirmationPanelProps.confirmationMsg}</p>
            </ModalBody>,
        )).toBeTruthy();
    });

    it('Should render the error correctly', () => {
        expect(component.contains(
            <ModalSubtitleAndError error={confirmationPanelProps.error}/>,
        )).toBeTruthy();
    });

    it('Should render the footer correctly', () => {
        expect(component.contains(
            <ModalFooter
                onConfirm={confirmHandler}
                onHide={hideHandler}
                confirmBtnText='Delete'
                cancelBtnText='Cancel'
                confirmDisabled={confirmationPanelProps.loading}
                cancelDisabled={confirmationPanelProps.loading}
                className='padding-v-12 padding-h-15'
            />,
        )).toBeTruthy();
    });
});
