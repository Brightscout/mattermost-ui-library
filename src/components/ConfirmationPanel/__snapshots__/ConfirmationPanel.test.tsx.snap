// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPanel Should render correctly 1`] = `
<div
  className="confirmation-panel primary-panel confirmation-dialog-class"
>
  <ModalBody>
    <p
      className="margin-top-10 margin-h-0 margin-bottom-15"
    >
      Dialog Message
    </p>
  </ModalBody>
  <ModalSubtitleAndError
    error="Dialog error"
  />
  <ModalFooter
    cancelBtnText="Cancel"
    cancelDisabled={true}
    className="padding-v-12 padding-h-15"
    confirmBtnText="Delete"
    confirmDisabled={true}
    onConfirm={[MockFunction]}
    onHide={[MockFunction]}
  />
</div>
`;
